src/algebra/group/inj_surj.lean

```diff
@@ -215,6 +225,16 @@ protected def semigroup [semigroup M₁] (f : M₁ → M₂) (hf : surjective f)
 { mul_assoc := hf.forall₃.2 $ λ x y z, by simp only [← mul, mul_assoc],
   ..‹has_mul M₂› }
 
+/-- A type endowed with `0` and `*`is a `semigroup_with_zero`,
+if it admits a surjective map that preserves `0` and `*` from a `semigroup_with_zero`. -/
+protected def semigroup_with_zero
+  [semigroup_with_zero M₁] [has_zero M₂] (f : M₁ → M₂) (hf : surjective f)
+  (zero : f 0 = 0) (mul : ∀ x y, f (x * y) = f x * f y) :
+  semigroup_with_zero M₂ :=
+{ zero_mul := hf.forall.2 $ λ x, by erw [← zero, ← mul, zero_mul],
```

Comment:
You should be able to use `..` on the mul_zero_class here, instead of proving these two fields

Reply:
Agreed but `function.injective.mul_zero_class` is in `algebra/group_with_zero/basic.lean` which is not imported. I'm reluctant to change imports just for this, especially as it may be that the right solution is to address the division of such lemmas mentioned in my comment [below].

Reply:
Oh I guess maybe you're saying I should just move the injective / surjective lemmas for `semigroup_with_zero` for this very reason.

Reply:
OK now done.